Laptop work

omriyaacobi opened the pull request:
ok i think im done with the add recipe page i messed around a bit with the css and stuff give stuff i did pourly so ill learn and im starting to work on the home page. :D

omriyaacobi committed:
Implement AddRecipe component with ingredient and instruction management; update routing in App and pass recipeList to Home

omriyaacobi committed:
Add styling and structure to AddRecipe component; implement ingredient and instruction management with popup confirmation

omriyaacobi committed:
Refactor AddRecipe component; extract AddIngredients and AddInstructions into separate components, enhance styling and layout for better user experience

YuvalShvartsman commented on src/AddRecipe/Components/AddIngredients.jsx:
```diff
@@ -0,0 +1,57 @@
+const AddIngredients = ({ setIngredientList }) => {
+  const [ingredientName, setIngredientName] = useState("");
+  const [ingredientQuantity, setIngredientQuantity] = useState("");
+  const ingredientRef = useRef(null);
```
I'd add an empty lione between the useRef hook and the rest of the states.

YuvalShvartsman commented on src/AddRecipe/Components/AddIngredients.jsx:
```diff
@@ -0,0 +1,57 @@
+            value={ingredientName}
+            onChange={(e) => setIngredientName(e.target.value)}
+            onKeyDown={(e) =>
+              e.key === "Enter" && (e.preventDefault(), addIngredient())
```
id also move this into a small handler, especially when this logic is written twice

YuvalShvartsman commented on src/AddRecipe/Components/AddIngredients.jsx:
```diff
@@ -0,0 +1,57 @@
+          value={ingredientQuantity}
+          onChange={(e) => setIngredientQuantity(e.target.value)}
+          onKeyDown={(e) =>
+            e.key === "Enter" && (e.preventDefault(), addIngredient())
```
This is the second time.

YuvalShvartsman commented on src/AddRecipe/Components/AddInstructions.jsx:
```diff
@@ -0,0 +1,32 @@
+        value={instruction}
+        onChange={(e) => setInstruction(e.target.value)}
+        onKeyDown={(e) =>
+          e.key === "Enter" && (e.preventDefault(), addInstruction())
```
again, small handler function looks cleaner as an end result

YuvalShvartsman commented on src/AddRecipe/Components/AddInstructions.jsx:
```diff
@@ -0,0 +1,32 @@
+  const [instruction, setInstruction] = useState("");
+
+  const addInstruction = () => {
+    if (!instruction.trim()) return;
```
"return" is no feedback for a user. + i dont really like stuff like if (!instruction.trim()).
I'd put in in a variable like - const cleanedInstruction = ins.trim...(not really sure about the name)
and also "negative" conditionals that start with ! are usually worse practice, try using positive variables.

omriyaacobi replied:
if you look at the page you'd understand why the return has no feedback i think its better like this its just so that you cant press the add instruction if its empty no its very convenient like that no need for extras.

YuvalShvartsman commented on src/AddRecipe/Components/AddIngredients.jsx:
```diff
@@ -0,0 +1,57 @@
+  const ingredientRef = useRef(null);
+
+  const addIngredient = () => {
+    if (!ingredientName.trim() || !ingredientQuantity.toString().trim()) return;
```
i wrote a comment about ifs like this in the next file + the return

YuvalShvartsman commented on src/AddRecipe/Components/AddIngredients.jsx:
```diff
@@ -0,0 +1,57 @@
+      {
+        ingredientName: ingredientName.trim(),
+        ingredientQuantity: ingredientQuantity.toString().trim(),
+      },
```
id move to an object before assigning it  - const newIngredient = {}
setIngredientList [...preve,newIngredient]

YuvalShvartsman commented on src/AddRecipe/AddRecipe.css:
```diff
@@ -0,0 +1,125 @@
+body {
+  background-color: #fff3e0; /* soft pastel orange */
```
maybe create some kind of a theme instead of writing  comments on colors.
you can reuse color scheme that you set for your entire application.
also, there is a vscode extention called "Color Highlight" that shows you the colors of the hexa shit.
so you can remove comments like these.
another quick tip while i am at it, there is another very nice extention caleld "turbo console.log) which is very helpful in my opinion, you should give it a look

YuvalShvartsman commented on src/AddRecipe/AddRecipe.css:
```diff
@@ -0,0 +1,125 @@
+  margin: 0;
+  font-family: Arial, sans-serif;
+}
+/* Container */
```
comments = weak ass code.

omriyaacobi replied:
i need an explanation on self explanatory code on the css front. i tried asking chat as well but it said css is common to use comments

YuvalShvartsman commented on src/AddRecipe/AddRecipe.css:
```diff
@@ -0,0 +1,125 @@
+.add-recipe {
+  width: 95%;
+  max-width: 1350px;
+  margin: 2rem 0 2rem 2%;   /* left-anchored, slight gutter */
```
gpt comments = even weaker ass code

YuvalShvartsman commented on src/AddRecipe/AddRecipe.css:
```diff
@@ -0,0 +1,125 @@
+  box-shadow: 0 4px 14px rgba(0,0,0,0.08);
+  text-align: left;
+  font-family: Arial, sans-serif;
+  box-sizing: border-box;    /* safer sizing baseline */
```
gpt comment club

YuvalShvartsman commented on src/AddRecipe/AddRecipe.css:
```diff
@@ -0,0 +1,125 @@
+  box-sizing: border-box;    /* safer sizing baseline */
+}
+
+/* Inputs (scoped—avoid global leakage) */
```
shut your ass

YuvalShvartsman commented on src/AddRecipe/AddRecipe.css:
```diff
@@ -0,0 +1,125 @@
+  box-shadow: 0 0 3px #2563eb55;
+}
+
+/* Headings / section titles */
```
remove

YuvalShvartsman commented on src/AddRecipe/AddRecipe.css:
```diff
@@ -0,0 +1,125 @@
+  font-size: 1.1rem;
+}
+
+/* Forms */
```
gay shit

YuvalShvartsman commented on src/AddRecipe/AddRecipe.css:
```diff
@@ -0,0 +1,125 @@
+}
+.ingredients-input {
+  display: flex;
+  gap: 0.5rem;       /* keeps them close but not glued */
```
gpt comment

YuvalShvartsman commented on src/AddRecipe/AddRecipe.css:
```diff
@@ -0,0 +1,125 @@
+}
+
+#ingredient {
+  flex: 1;           /* fills available space */
```
just remove all comments ffs

YuvalShvartsman commented on src/AddRecipe/AddRecipe.jsx:
```diff
@@ -1,4 +1,103 @@
-const AddRecipe = () => {
-  return <p>nigga</p>;
+import { useState } from "react";
+import "./AddRecipe.css";
```
order your imports , first react related , then css , then components.
lines of space before each group

YuvalShvartsman commented on src/AddRecipe/AddRecipe.jsx:
```diff
@@ -1,4 +1,103 @@
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    if (!name.trim() || !ingredientList.length || !instructionList.length) {
```
if is no good

YuvalShvartsman commented on src/AddRecipe/AddRecipe.jsx:
```diff
@@ -1,4 +1,103 @@
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    if (!name.trim() || !ingredientList.length || !instructionList.length) {
+      alert(
```
nice alert but i bet its fucking ugly , create your own

YuvalShvartsman commented on src/AddRecipe/AddRecipe.jsx:
```diff
@@ -1,4 +1,103 @@
+      );
+      return;
+    }
+    const recipe = {
```
at least here you did move it into an object, nice!

YuvalShvartsman commented on src/AddRecipe/AddRecipe.jsx:
```diff
@@ -1,4 +1,103 @@
+    setPreparationTime("");
+    setIngredientList([]);
+    setInstructionList([]);
+    setShowPopup(true);
```
i'd move all of these into a small function called "resetFields" or something like that

YuvalShvartsman commented on src/AddRecipe/AddRecipe.jsx:
```diff
@@ -1,4 +1,103 @@
+
+  return (
+    <div className="add-recipe">
+      {/* SINGLE form wraps everything */}
```
gpt comment = gay

omriyaacobi replied:
are co pilot comments ok ?

YuvalShvartsman commented on src/AddRecipe/AddRecipe.jsx:
```diff
@@ -1,4 +1,103 @@
+          <div className="popup">
+            <h3>Recipe saved!</h3>
+            <p>Your recipe has been added successfully.</p>
+            <button onClick={() => navigate("/")}>Go to Home</button>
```
even the navigate you can move into a small one liner function handler for cleaner jsx

YuvalShvartsman commented on src/App.jsx:
```diff
@@ -3,15 +3,26 @@ import Home from "./Home/Home.jsx";
+
   return (
     <>
       <div>
```
unneccessary div

YuvalShvartsman commented on src/App.jsx:
```diff
@@ -3,15 +3,26 @@ import Home from "./Home/Home.jsx";
 import About from "./About.jsx";
 import AddRecipe from "./AddRecipe/AddRecipe.jsx";
 import { Routes, Route, Link } from "react-router-dom";
+import { useState } from "react";
```
order imports

omriyaacobi committed:
Enhance AddRecipe component functionality; implement field reset after adding a recipe, improve ingredient and instruction input handling, and add styling to Home component

Commit:
Refactor AddRecipe component; improve input validation logic, enhance styling comments, and adjust import order for better readability

Commit:
Enhance AddRecipe component; add meal type selection, image upload functionality, and improve form validation. Refactor Home component to display recipes using RecipeCard with responsive design.

Commit:
Add recipe data, Navbar component, and styles

- Created RecipesData.js with a collection of recipes including details like name, meal types, preparation time, ingredients, and instructions.
- Implemented Navbar component with navigation links for different meal categories and an "Add Recipe" button.
- Added CSS styles for the Navbar to ensure a fixed position, responsive layout, and hover effects for links.

Commit:
Refactor CSS styles for background image and navbar; adjust z-index and remove unnecessary properties for improved layout consistency. Update Home component styles for better visual hierarchy and spacing.

Commit:
Refactor AddRecipe component structure and styles

- Moved AddRecipe, AddIngredients, and AddInstructions components to a new directory structure under `src/components/AddRecipe/`.
- Deleted old CSS and JSX files for AddRecipe, AddIngredients, and AddInstructions.
- Created new CSS and JSX files for AddRecipe, AddIngredients, and AddInstructions with updated styles and structure.
- Updated Navbar and RecipeCard components to reflect new paths and styles.
- Removed unused CSS files for Navbar and RecipeCard.
- Enhanced Home component layout and styles for better user experience.

Commit:
Add CategoryPages component and integrate with routing; refactor meal type options to lowercase and update Home component to use CardGrid for recipe display.

omriyaacobi committed:
Add RecipePage component and integrate with routing; update recipe data structure to include unique IDs and adjust CardGrid links accordingly.

omriyaacobi committed:
Add RecipeNotFound component and integrate into RecipePage and CardGrid; refactor styles for consistency and improve layout across components.

Commit:
Refactor App component structure and improve routing integration

Commit:
Refactor Home component and update welcome message for clarity
